### PR TITLE
110 add missing events

### PR DESCRIPTION
**Pull request by lirona**

### CODE REVIEW

File: contracts/HATVaults.sol [hunk 1]
```diff
@@ -280,29 +281,31 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
-    event ClaimSubmitted(uint256 indexed _pid,
+    event SubmitClaim(uint256 indexed _pid,
                             address indexed _beneficiary,
                             uint256 indexed _severity,
                             address _committee);
 
-    event WithdrawRequest(uint256 indexed _pid,
-                        address indexed _beneficiary,
+    event WithdrawRequest(address indexed _beneficiary,
+                        uint256 indexed _pid,
```
[the remark this answers is not in the document]
Comment (jellegerbrandy) [reply contracts/HATVaults.sol:296]: I mean, SubmitClaim just above has the _pid first

File: contracts/HATVaults.sol [hunk 2]
```diff
@@ -280,29 +281,31 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
-    
     event SetClaimFee(uint256 _fee);
-
-    event RewardDepositors(uint256 indexed _pid, uint256 indexed _amount);
+    event RewardDepositors(address indexed user, uint256 indexed pid, uint256 indexed amount);
+    event DepositHATReward(address indexed user, uint256 indexed amount);
+    event ClaimReward(address indexed user, uint256 indexed pid);
+    event UpdatePool(uint256 indexed pid, uint256 indexed rewardPerShare);
+    event SetWithdrawRequestParams(uint256 indexed withdrawRequestPendingPeriod, uint256 indexed withdrawRequestEnablePeriod);
+    event DismissClaim(uint256 indexed pid, address indexed sender);
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:307]: see earlier comments on adding `msg.sender` to events

File: contracts/HATVaults.sol [hunk 3]
```diff
@@ -400,6 +406,7 @@ contract  HATVaults is Governable, ReentrancyGuard {
         pool.rewardPerShare = pool.rewardPerShare + (reward * 1e12 / totalShares);
         pool.lastRewardBlock = block.number;
         pool.lastProcessedTotalAllocPoint = lastPoolUpdate;
+        emit UpdatePool(_pid, pool.rewardPerShare);
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:409]: so one reason against this event here is that updatepool is called on each deposit and withdrawal, which already have events..

File: contracts/HATVaults.sol [hunk 4]
```diff
@@ -1138,26 +1133,13 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
-        claimBounty.swapAndBurn =
-        totalBountyAmount * bountyInfos[_pid].bountySplit.swapAndBurn
-        / (HUNDRED_PERCENT * HUNDRED_PERCENT);
-        claimBounty.governanceHat =
-        totalBountyAmount * bountyInfos[_pid].bountySplit.governanceHat
-        / (HUNDRED_PERCENT * HUNDRED_PERCENT);
-        claimBounty.hackerHat =
-        totalBountyAmount * bountyInfos[_pid].bountySplit.hackerHat
-        / (HUNDRED_PERCENT * HUNDRED_PERCENT);
+        uint256 totalBountyAmount = totalSupply * bountyInfos[_pid].bountyLevels[_severity] / HUNDRED_PERCENT;
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:1136]: this change actually changes the rounding behavior, so now things are less precise. Perhaps not significant, but still better not to change this

File: contracts/HATVaults.sol [hunk 5]
```diff
@@ -938,32 +884,46 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
-   * @param _registered does this pool is registered (default true).
+   * @param _visable is this pool visable in the UI
    * @param _depositPause pause pool deposit (default false).
    * This parameter can be used by the UI to include or exclude the pool
    * @param _descriptionHash the hash of the pool description.
  */
     function setPool(uint256 _pid,
                     uint256 _allocPoint,
-                    bool _registered,
+                    bool _visable,
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:894]: `_visible`

File: contracts/HATVaults.sol [hunk 6]
```diff
@@ -938,32 +884,46 @@ contract  HATVaults is Governable, ReentrancyGuard {
    * @dev setPool
    * @param _pid the pool id
    * @param _allocPoint the pool allocation point
-   * @param _registered does this pool is registered (default true).
+   * @param _visable is this pool visable in the UI
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:887]: `_visible`

File: contracts/HATVaults.sol [hunk 7]
```diff
@@ -244,12 +245,12 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
     event Deposit(address indexed user, uint256 indexed pid, uint256 amount);
-    event Withdraw(address indexed user, uint256 indexed pid, uint256 amount);
+    event Withdraw(address indexed user, uint256 indexed pid, uint256 shares);
     event EmergencyWithdraw(address indexed user, uint256 indexed pid, uint256 amount);
-    event SendReward(address indexed user, uint256 indexed pid, uint256 amount, uint256 requestedAmount);
+    event SafeTransferReward(address indexed user, uint256 indexed pid, uint256 amount, uint256 requestedAmount);
     event MassUpdatePools(uint256 _fromPid, uint256 _toPid);
 
-    event SetCommittee(uint256 indexed _pid, address indexed _committee);
+    event SetCommittee(uint256 indexed _pid, address indexed _committee, address indexed sender);
```
[the remark this answers is not in the document]
Comment (lirona) [reply contracts/HATVaults.sol:253]: oh ok makes sense

File: contracts/HATVaults.sol (hunk 2 above)
Comment (lirona) [reply contracts/HATVaults.sol:307]: removed them all

File: contracts/HATVaults.sol (hunk 3 above)
Comment (lirona) [reply contracts/HATVaults.sol:409]: i was thinking maybe we want to know what happened (the new reward per share)

File: contracts/HATVaults.sol [hunk 8]
```diff
@@ -1127,15 +1120,15 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
-    function poolLength() external view returns (uint256) {
+    function getNumberOfPools() external view returns (uint256) {
         return poolInfos.length;
     }
 
     function calcClaimBounty(uint256 _pid, uint256 _severity)
     public
     view
     returns(ClaimBounty memory claimBounty) {
-        uint256 totalSupply = poolInfos[_pid].balance;
```
Comment (ben-kaufman) [inline contracts/HATVaults.sol:1138]: Why is this not aligned now properly

**Review by jellegerbrandy — CHANGES_REQUESTED** [review]
I guess if you address ben's comments it is good to go :)

File: contracts/HATVaults.sol [hunk 9]
```diff
@@ -262,47 +263,47 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
 
-    event ClaimSubmitted(uint256 indexed _pid,
+    event SubmitClaim(uint256 indexed _pid,
                             address indexed _beneficiary,
-                            uint256 indexed _severity,
-                            address _committee);
+                            uint256 indexed _severity);
 
     event WithdrawRequest(uint256 indexed _pid,
-                        address indexed _beneficiary,
```
[the remark this answers is not in the document]
Comment (lirona) [reply contracts/HATVaults.sol:296]: because it's msg.sender

File: contracts/HATVaults.sol [hunk 10]
```diff
@@ -262,47 +263,47 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
                         uint256 indexed _withdrawEnableTime);
 
     event SetWithdrawSafetyPeriod(uint256 indexed _withdrawPeriod, uint256 indexed _safetyPeriod);
-    
     event SetRewardMultipliers(uint256[24] _rewardMultipliers);
-    
     event SetClaimFee(uint256 _fee);
-
-    event RewardDepositors(uint256 indexed _pid, uint256 indexed _amount);
+    event RewardDepositors(uint256 indexed pid, uint256 indexed amount);
```
[the remark this answers is not in the document]
Comment (lirona) [reply contracts/HATVaults.sol:300]: yeah, i just didn't want to make a mess. it will be part of fixing #143

**Comment by lirona** [comment]
thanks @ben-kaufman for reviewing my pr!
you guys haven't answered my comments though.. @jellegerbrandy

File: contracts/HATVaults.sol (hunk 9 above)
[the remark this answers is not in the document]
Comment (jellegerbrandy) [reply contracts/HATVaults.sol:296]: just keep it, it has some puprose here bc it is indexed

File: contracts/HATVaults.sol (hunk 10 above)
[the remark this answers is not in the document]
Comment (jellegerbrandy) [reply contracts/HATVaults.sol:300]: so I do not see where the style guide says anything about initial underscores.

I say to add them to be consistent with (almost) all other events where the args start with underscores, no need to merge things we already know how we want to change

File: contracts/HATVaults.sol [hunk 11]
```diff
@@ -262,47 +263,47 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
+    event DepositHATReward(uint256 indexed amount);
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:301]: We need tests for each of the new events

File: contracts/HATVaults.sol [hunk 12]
```diff
@@ -262,47 +263,47 @@ contract  HATVaults is Governable, ReentrancyGuard {
...
+                    uint256 indexed _amountSwapped,
                     uint256 _amountReceived,
                     address _tokenLock);
 
-    event SwapAndBurn(uint256 indexed _pid, uint256 indexed _amountSwaped, uint256 indexed _amountBurned);
+    event SwapAndBurn(uint256 indexed _pid, uint256 indexed _amountSwapped, uint256 indexed _amountBurned);
     event SetVestingParams(uint256 indexed _pid, uint256 indexed _duration, uint256 indexed _periods);
     event SetHatVestingParams(uint256 indexed _duration, uint256 indexed _periods);
 
-    event ClaimApproved(address indexed _committee,
```
Comment (jellegerbrandy) [inline contracts/HATVaults.sol:283]: no wait, do not remove the indexed addresses, even if it is msg.sender (indexed means: you can search for them in the events log, which is useful)

**Review by jellegerbrandy — CHANGES_REQUESTED** [review]
hi liron, more requested changes

File: contracts/HATVaults.sol (hunk 10 above)
[the remark this answers is not in the document]
Comment (lirona) [reply contracts/HATVaults.sol:300]: it doesn't say anything against, but there's no reason there should be an underscore there.. i thought would be better to remove all underscores

**Pull request merged**